### PR TITLE
Sawfly code refactor 

### PR DESCRIPTION
**Pull request by NightmareChamillian**
## About the PR 
This PR cleans up a lot of the code for sawflies, ports them to their own DMI, and reworks their grenade code to not create and destroy new sawflies.

## Why's this needed? 
Removes jank, makes code better.

## Changelog
```
(u) NightmareChamillian
(+) Sawflies have had some significant changes to their grenade/deployment code. Please report any bugs.
```

### CODE REVIEW

**Comment by ZWRh3** [comment]
this is mega minor but i think you meant to say refactor not refractor

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Not interested in adding playable sawflies

**Comment by NightmareChamillian** [comment]
@Tarmunora This PR doesn't make sawflies any more or less playable than they are when they were first merged. Them being player controlled is still admin-only, I've just gone through and added code so that if you are controlling them, you can actually do things.

**Comment by Tarmunora** [comment]
To clarify. We're not interested in having support for playable sawflies, now or in the future, so adding framework for it is needless bloat.

**Comment by NightmareChamillian** [comment]
Of course! I'll be sure to gut that.

File: code/obj/sawflymisc.dm [hunk 1]
```diff
@@ -17,58 +17,62 @@
...
-	mechanics_type_override = /obj/item/old_grenade/sawfly
-	issawfly = TRUE
-	contraband = 2
-	//var/tempdam = 0
+	//used in dictating behavior when deployed from grenade
 	var/mob/living/critter/robotic/sawfly/heldfly = null
+	var/obj/item/organ/brain/currentbrain = null
+	var/mob/currentuser = null
+	var/isopen = FALSE
+	var/playercontrolled = FALSE
```
Comment (Tarmunora) [inline code/obj/sawflymisc.dm:38]: also cut these bits

File: code/obj/sawflymisc.dm [hunk 2]
```diff
@@ -135,18 +139,15 @@
...
-
-
 // ---------------limb---------------
 /datum/limb/sawfly_blades
 
 	//due to not having intent hotkeys and also being AI controlled we only need the one proc
 	harm(mob/living/target, var/mob/living/critter/robotic/sawfly/user) //will this cause issues down the line when someone eventually makes a child of this? hopefully not
-		if(!ON_COOLDOWN(user, "sawfly_attackCD", 1 SECONDS))
-			user.visible_message("<b class='alert'>[user] [pick(list("gouges", "cleaves", "lacerates", "shreds", "cuts", "tears", "saws", "mutilates", "hacks", "slashes",))] [target]!</b>")
+		if(!ON_COOLDOWN(user, "sawfly_attackCD", 0.8 SECONDS))
```
Comment (Tarmunora) [inline code/obj/sawflymisc.dm:147]: ?

File: code/obj/sawflymisc.dm [hunk 3]
```diff
@@ -17,58 +17,62 @@
 	desc = "A self-deploying antipersonnel robot. It's folded up and offline..."
 	det_time = 1.5 SECONDS
 	throwforce = 7
+	icon = 'icons/obj/items/sawfly.dmi'
```
Comment (Tarmunora) [inline code/obj/sawflymisc.dm:20]: cut icon_states from old icon file here too

File: code/obj/sawflymisc.dm (hunk 2 above)
Comment (NightmareChamillian) [reply code/obj/sawflymisc.dm:147]: Old code from pre-nerf I missed that shouldn't be there, good catch.

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**